engineering/conventions.md

```diff
@@ -79,6 +80,18 @@ Your application will expose one or more binaries with a CLI.
 We prefer [GNU-style command line options](http://www.catb.org/esr/writings/taoup/html/ch10s05.html).
 That is, double dash for long options and single dash for short options (single letter). Include `--help` and `--version`. As a general rule, define always long flags, and then optionally short flags for tools that are often used interactively. When defining flags it is worth to consider matching behavior of widely used flags. For example, if `--all` is present, `-a` should be its shorthand. `--quiet` as a default instead of `--silent`. `--file/-f` for a file argument, `--output/-o` for output path, etc.
 
+## Logging
+
+* Use structured logging.
+* By default, output pretty logs when running in a terminal, output JSON when not.
+* Allow changing logging level and format explicitly through environment variables (`LOG_LEVEL`, `LOG_FORMAT`).
+* Use the following keys where they apply:
+  * `time` for log timestamp.
+  * `level` for log level.
+  * `error` for error message (optionally with stacktrace).
+  * `msg` for main log message.
+  * `source` for source file originating the log message (e.g. `foo.go:32`).
```

Comment:
I would add an optional key to log the time it took to do something so all the tools use the same one. Currently borges uses `duration` and the value has a unit, for example: `145.7622384ms`.

Reply:
I have to check which formats are compatible with Kibana.

Reply:
I see borges uses plain integer for duration when setup with the JSON format (e.g. `"duration":3551331`). This is ok and we just have to agree on the units. Then we can just specify the units in Kibana, but they need to be the same units for all projects.

Reply:
Added duration field, units are nanoseconds. This is the default in Go and it should be ok for all our applications.